internal/controllers/machine/machine_controller_noderef.go

```diff
@@ -109,7 +110,11 @@ func (r *Reconciler) reconcileNode(ctx context.Context, cluster *clusterv1.Clust
 		desired[clusterv1.OwnerKindAnnotation] = owner.Kind
 		desired[clusterv1.OwnerNameAnnotation] = owner.Name
 	}
-	if annotations.AddAnnotations(node, desired) {
+	annotationsHaveChanged := annotations.AddAnnotations(node, desired)
+
+	managedLabelsHaveChanged := reconcileManagedLabels(machine, node)
+
+	if annotationsHaveChanged || managedLabelsHaveChanged {
```

Comment:
I think we can avoid this check given that patchHelper already implements a logic comparing initial/final state and it applies the patch only if there are differences